webserver/src/main/java/io/kestra/webserver/controllers/NamespaceFileController.java

```diff
@@ -109,6 +116,23 @@ public void delete(
         storageInterface.delete(tenantService.resolveTenant(), toNamespacedStorageUri(namespace, path));
     }
 
+    @ExecuteOn(TaskExecutors.IO)
+    @Get(uri = "files/distinct-namespaces", produces = MediaType.TEXT_JSON)
+    @Operation(tags = {"Files"}, summary = "List existing namespace folders")
+    public List<String> distinctNamespaces() throws IOException, URISyntaxException {
+        List<String> storageNamespaces =  distinctNamespacesFiles();
+        List<String> flowNamespaces = flowRepository.findDistinctNamespace(tenantService.resolveTenant());
+
+        return Stream.concat(storageNamespaces.stream(), flowNamespaces.stream()).distinct().toList();
+    }
```

Comment:
as in the comment of the issue, I'm not sure the merge should be done at the backend level as it might be interesting for admin purpose to be able to list exclusively namespaces with files

Reply:
Yes, those are two very different things: flows and namespace files, and we want to display both of them in the editor. So it's the responsibility of the editor to merge them.